addOns/custompayloads/src/main/java/org/zaproxy/zap/extension/custompayloads/CustomPayloadMultipleOptionsTableModel.java

```diff
@@ -63,6 +63,24 @@ public void resetToDefaults() {
         }
     }
 
+    public void addToTable(CustomPayload payload, boolean checkDuplicates) {
```

Comment:
Better deduplicate/process the payloads beforehand, here it should just add. Also, better to collect the payloads once rather than for each time a payload is added.

Reply:
The latter part was not yet addressed.